run_training.py

```diff
@@ -0,0 +1,1399 @@
+        mean = channel_mean[c]
+        std = channel_std[c]
+        channel_slice = (input_image[:, c, ...] - mean) / \
+                        (std + np.finfo(float).eps)
+        # channel_slice = t.clamp(channel_slice, -1, 1)
+        channel_slices.append(channel_slice)
+    norm_img = np.stack(channel_slices, 1)
+    # norm_img = (input_image - mean.astype(np.float64)) /\
+    #            (std + np.finfo(float).eps)
+    return norm_img
+
+
+def unzscore(im_norm, mean, std):
+    """
+    Revert z-score normalization applied during preprocessing. Necessary
+    before computing SSIM
+
+    :param input_image: input image for un-zscore
+    :return: image at its original scale
+    """
+
+    im = im_norm * (std + np.finfo(float).eps) + mean
+
+    return im
+
+def rescale(dataset):
+    """ Rescale value range of image patches in `dataset` to CHANNEL_RANGE
+
+    Args:
+        dataset (TensorDataset): dataset before rescaling
+
+    Returns:
+        TensorDataset: dataset after rescaling
+
+    """
+    tensor = dataset.tensors[0]
+    channel_mean = t.mean(tensor, dim=[0, 2, 3])
+    channel_std = t.mean(tensor, dim=[0, 2, 3])
+    print('channel_mean:', channel_mean)
+    print('channel_std:', channel_std)
+    assert len(channel_mean) == tensor.shape[1]
+    channel_slices = []
+    for i in range(len(CHANNEL_RANGE)):
+        mean = channel_mean[i]
+        std = channel_std[i]
+        channel_slice = (tensor[:, i] - mean) / std
+        # channel_slice = t.clamp(channel_slice, -1, 1)
+        channel_slices.append(channel_slice)
+    new_tensor = t.stack(channel_slices, 1)
+    return TensorDataset(new_tensor)
+
+
+def resscale_backward(tensor):
+    """ Reverse operation of `rescale`
+
+    Args:
+        dataset (TensorDataset): dataset after rescaling
+
+    Returns:
+        TensorDataset: dataset before rescaling
+
+    """
+    assert len(tensor.shape) == 4
+    assert len(CHANNEL_RANGE) == tensor.shape[1]
+    channel_slices = []
+    for i in range(len(CHANNEL_RANGE)):
+        lower_, upper_ = CHANNEL_RANGE[i]
+        channel_slice = lower_ + tensor[:, i] * (upper_ - lower_)
+        channel_slices.append(channel_slice)
+    new_tensor = t.stack(channel_slices, 1)
+    return new_tensor
+
+def concat_relations(relations, offsets):
+    """combine relation dictionaries from multiple datasets
+
+    Args:
+        relations (list): list of relation dict to combine
+        offsets (list): offset to add to the indices
+
+    Returns: new_relations (dict): dictionary of combined relations
+
+    """
+    new_relations = {}
+    for relation, offset in zip(relations, offsets):
+        old_keys = relation.keys()
+        new_keys = [(id1 + offset, id2 + offset) for id1, id2 in old_keys]
+        # make a new dict with updated keys
+        relation = dict(zip(new_keys, relation.values()))
+        new_relations.update(relation)
+
+    return new_relations
+
+
+
+if __name__ == '__main__':
```

Comment:
I think it will be great to write a simple CLI for training as well (I could help with it once I get the data ready).

Reply:
I have already started a `CLI` branch that parses a .yml config file.  This will make it easy to attach hyperparams to a given training run.  Please take a look ... it will need to merge with this branch after this one is finished.